keanu-python/tests/test_sampling.py

```diff
@@ -98,6 +193,32 @@ def test_can_iter_through_samples(algo: Callable[[BayesNet], PosteriorSamplingAl
     count = 0
     for sample in islice(samples, draws):
         count += 1
+
+    assert count == draws
+
+
+@pytest.mark.parametrize(
+    "algo", [(lambda net: MetropolisHastingsSampler(proposal_distribution="prior", latents=net.get_latent_vertices())),
+             (lambda net: NUTSSampler())])
+def test_can_iter_through_tensor_samples(algo: Callable[[BayesNet], PosteriorSamplingAlgorithm],
+                                         tensor_net: BayesNet) -> None:
+    draws = 10
+    samples = generate_samples(
+        net=tensor_net,
+        sample_from=tensor_net.get_latent_vertices(),
+        sampling_algorithm=algo(tensor_net),
+        down_sample_interval=1)
+    count = 0
+    for sample in islice(samples, draws):
+        count += 1
+        assert ('exp', '(0, 0)') in sample
+        assert ('exp', '(0, 1)') in sample
+        assert ('exp', '(1, 0)') in sample
+        assert ('exp', '(1, 1)') in sample
+        assert ('cauchy', '(0, 0)') in sample
+        assert ('cauchy', '(0, 1)') in sample
+        assert ('cauchy', '(1, 0)') in sample
+        assert ('cauchy', '(1, 1)') in sample
```

Comment:
for distribution in ('exp', 'cauchy'):
        for i in (0, 1):
            for j in (0, 1):
                assert((distribution, '({}, {})'.format(i, j)) in sample)